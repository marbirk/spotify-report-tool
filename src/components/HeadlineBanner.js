import React from 'react'
import Typewriter from 'typewriter-effect'
import { Link } from 'gatsby'

const HeadlineBanner = props => {
    const isStartPage = props.pathname.length === 1
    return (
        <section className="mb-12 mt-6">
            <Link to="/">
                <h1 className="squishy-text">
                    {isStartPage ? getStartPageBanner() : getDefaultBanner()}
                </h1>
            </Link>
        </section>
    )
}

const getStartPageBanner = () => {
    return (
        <>
            <span>I am</span>
            <Typewriter
                options={{
                    strings: [
                        'Marcel Birkhahn',
                        'Frontend Engineer',
                        'Informatics Student',
                        'Hard Rock Enthusiast',
                        'Festival Fan',
                    ],
                    autoStart: true,
                    loop: true,
                }}
            />
        </>
    )
}

const getDefaultBanner = () => {
    const getNameString = name => <span className="block">{name}</span>
    return (
        <>
            {getNameString('Marcel')}
            {getNameString('Birkhahn')}
        </>
    )
}

export default HeadlineBanner
