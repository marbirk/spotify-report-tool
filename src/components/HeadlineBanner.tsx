import React from 'react'
import Typewriter from 'typewriter-effect'
import { Link } from 'gatsby'

interface HeadlineBannerProps {
    pathname: string
}

const HeadlineBanner = ({ pathname }: HeadlineBannerProps) => {
    const isStartPage = pathname.length === 1
    return (
        <div className="mb-32 mt-24">
            <Link to="/">
                <h1 className="u-squishy-text">
                    {isStartPage ? getStartPageBanner() : getDefaultBanner()}
                </h1>
            </Link>
        </div>
    )
}

const getStartPageBanner = () => {
    return (
        <>
            <span>I am</span>
            <Typewriter
                onInit={() => {}}
                options={{
                    strings: [
                        'Marcel Birkhahn',
                        'Frontend Engineer',
                        'Studying IT Master',
                        'Hard Rock Fan',
                        'Nature Lover',
                    ],
                    autoStart: true,
                    loop: true,
                }}
            />
        </>
    )
}

const getDefaultBanner = () => {
    const getNameString = (name: string) => (
        <span className="block">{name}</span>
    )
    return (
        <>
            {getNameString('Marcel')}
            {getNameString('Birkhahn')}
        </>
    )
}

export default HeadlineBanner
