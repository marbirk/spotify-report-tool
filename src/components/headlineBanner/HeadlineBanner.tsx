import React from 'react'
import Typewriter from 'typewriter-effect'
import Link from '../link/Link'

interface HeadlineBannerProps {
    pathname: string
}

const HeadlineBanner = ({ pathname }: HeadlineBannerProps) => {
    const isStartPage = pathname === '/'
    return (
        <div className="mb-32 mt-24" data-testid="headline-banner">
            <Link to="/">
                <h1 className="u-squishy-text">
                    {isStartPage ? getStartPageBanner() : getDefaultBanner()}
                </h1>
            </Link>
        </div>
    )
}

const getStartPageBanner = () => {
    return (
        <>
            <span>I am</span>
            <Typewriter
                onInit={typewriter => {
                    typewriter
                        .pauseFor(2000)
                        .typeString('Marcel Birkhahn')
                        .deleteAll()
                        .typeString('Frontend Engineer')
                        .deleteAll()
                        .typeString('Studying IT Master')
                        .deleteAll()
                        .typeString('Hard Rock Fan')
                        .pauseFor(1000)
                        .start()
                }}
                options={{
                    loop: true,
                }}
            />
        </>
    )
}

const getDefaultBanner = () => {
    const getNameString = (name: string) => (
        <span className="block">{name}</span>
    )
    return (
        <>
            {getNameString('Marcel')}
            {getNameString('Birkhahn')}
        </>
    )
}

export default HeadlineBanner
