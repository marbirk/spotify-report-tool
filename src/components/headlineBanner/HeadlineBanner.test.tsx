import React from 'react'
import { render } from '@testing-library/react'

import HeadlineBanner from './HeadlineBanner'

describe('HeadlineBanner', () => {
    test('renders on home correctly', () => {
        const { container, getByText } = render(<HeadlineBanner pathname="/" />)
        expect(getByText('I am')).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })
    test('renders on other pages correctly', () => {
        const { container, getByText } = render(
            <HeadlineBanner pathname="/about" />
        )
        expect(getByText('Marcel')).toBeInTheDocument()
        expect(container).toMatchSnapshot()
    })
})
