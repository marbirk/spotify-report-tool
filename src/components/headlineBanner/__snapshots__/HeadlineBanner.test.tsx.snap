// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadlineBanner renders on home correctly 1`] = `
<div
  class="mb-32 mt-24"
  data-testid="headline-banner"
>
  <a
    href="/"
  >
    <h1
      class="u-squishy-text"
    >
      Spotify report
    </h1>
  </a>
</div>
`;

exports[`HeadlineBanner renders on other pages correctly 1`] = `
<div
  class="mb-32 mt-24"
  data-testid="headline-banner"
>
  <a
    href="/"
  >
    <h1
      class="u-squishy-text"
    >
      Spotify report
    </h1>
  </a>
</div>
`;
